app/uk/gov/hmrc/transitmovements/services/DeparturesService.scala

```diff
@@ -35,7 +36,7 @@ import scala.concurrent.Future
 
 @ImplementedBy(classOf[DeparturesServiceImpl])
 trait DeparturesService {
-  def insertDeparture(eori: EORINumber, declarationData: DeclarationData): EitherT[Future, Throwable, DeclarationResponse]
+  def insertDeparture(eori: EORINumber, declarationData: DeclarationData): EitherT[Future, MongoError, DeclarationResponse]
```

Comment:
This name seems off (I think I've overlooked this before) - we don't insert a departure as we aren't asking for one to store. This should probably be createDeparture with createDeparture as an internal/private method or potentially even an Object method or apply overload?